src/main/java/org/typo3/solr/search/AccessFilter.java

```diff
@@ -87,76 +100,103 @@ public AccessFilter(final String field, final String userGroupList) {
   }
 
   /**
-   * Filters the documents based on the access granted.
+   * This method iterates over the documents and marks documents as accessable that are granted
+   * and have the access information in a single value field.
    *
-   * @param context @inheritDoc
-   * @param acceptDocs @inheritDoc
-   * @return OpenBitSet
-   * @throws IOException When an error occurs while reading from the index.
-   */
-  @Override
-  public final DocIdSet getDocIdSet(final LeafReaderContext context, final Bits acceptDocs) throws IOException {
-    LeafReader reader = context.reader();
-    FixedBitSet bits = new FixedBitSet(reader.maxDoc());
+   * @param doc
+   * @param values
+   * @throws IOException
+     */
 
-    DocValuesType type = reader.getFieldInfos().fieldInfo(accessField).getDocValuesType();
-    Boolean isMultivalue = type.equals(DocValuesType.SORTED_SET);
+  private boolean handleSingleValueAccessField(int doc, SortedDocValues values) throws IOException {
 
-    if(isMultivalue) {
-      handleMultivalueAccessField(reader, bits);
-    } else {
-      handleSingleValueAccessField(reader, bits);
-    }
+    BytesRef bytes = values.get(doc);
+    String documentGroupList = bytes.utf8ToString();
 
-    return new BitDocIdSet(bits);
+    if (accessGranted(documentGroupList)) {
+      return true;
+    }
+    return false;
   }
 
+
   /**
    * This method iterates over the documents and marks documents as accessable that are granted
    * and have the access information in a single value field.
    *
-   * @param reader
-   * @param bits
+   * @param doc
+   * @param multiValueSet
    * @throws IOException
      */
-  private void handleSingleValueAccessField(LeafReader reader, FixedBitSet bits) throws IOException {
-    SortedDocValues values = reader.getSortedDocValues(accessField);
+  private boolean handleMultivalueAccessField(int doc, SortedSetDocValues multiValueSet) throws IOException {
+    long ord;
+    multiValueSet.setDocument(doc);
 
-    for (int documentPointer = 0; documentPointer < reader.maxDoc(); documentPointer++) {
-      BytesRef bytes = values.get(documentPointer);
+    while ((ord = multiValueSet.nextOrd()) != SortedSetDocValues.NO_MORE_ORDS) {
+      BytesRef bytes = multiValueSet.lookupOrd(ord);
       String documentGroupList = bytes.utf8ToString();
 
       if (accessGranted(documentGroupList)) {
-        bits.set(documentPointer);
+	return true;
```

Comment:
This is missing JavaDoc, too. Just shortly describe under which condition it returns what... "Returns true if ..., false otherwise"